code/game/objects/items/extinguisher.dm

```diff
@@ -43,7 +45,19 @@
 /obj/item/extinguisher/New()
 	..()
 	create_reagents(max_water)
-	reagents.add_reagent("water", max_water)
+	reagents.add_reagent(chem, max_water)
+
+
+/obj/item/extinguisher/advanced
+	name = "advanced fire extinguisher"
+	desc = "Used to stop thermonuclear fires from spreading inside your engine."
+	icon_state = "foam_extinguisher0"
+	item_state = "foam_extimguisher"

```

Comment:
Typo and also does this have inhand sprites? If it doesn't you can use the normal extinguisher ones.

Reply:
where is the typo?

Reply:
"extimguisher"